CONTRIBUTING.md

```diff
@@ -15,52 +55,221 @@ simple statement that you, as a contributor, have the legal right to make the
+for a week before nudging us.
+
+Once you have created your issue, we recommend posting it in the slack channel
+to get more eyes on it faster.
+
+Please following these steps when opening a new help issue:
+
+1. **Search existing issues.** If something similar already exists, and is still open, please contribute to the discussion there.
+
+1. Write a title with the format "How to x".
+
+1. Explain what you are trying to accomplish, what you have tried, and the behaviour you are seeing.
+
+1. Please include your config (removing any sensitive information) or exact the commands you're using.
+   Please ensure everything is formatted nicely with [code blocks](https://docs.github.com/en/free-pro-team@latest/github/writing-on-github/creating-and-highlighting-code-blocks).
+
+1. When providing verbose logs, please use either a Gist or [HTML detail tags](https://gist.github.com/ericclemmons/b146fe5da72ca1f706b2ef72a20ac39d).
+
+---
+
+# Submitting PRs
+## Choosing something to work on
+
+If you are not here to report a bug, ask for help or request some new behaviour, this
+is the section for you. We have curated a set of issues for anyone who simply
+wants to build up their open-source cred :muscle:.
+
+- Issues labelled [`good first issue`](https://github.com/weaveworks/eksctl/labels/good%20first%20issue)
+  should be accessible to folks new to the repo, as well as to open source in general.
+
+  These issues should present a low/non-existent barrier to entry with a thorough description,
+  easy-to-follow reproduction (if relevant) and enough context for anyone to pick up.
+  The objective should be clear, possibly with a suggested solution or some pseudocode.
+  If anything similar has been done, that work should be linked.
+
+  If you have come across an issue tagged with `good first issue` which you think you would
+  like to claim but isn't 100% clear, please ask for more info! When people write issues
+  there is a _lot_ of assumed knowledge which is very often taken for granted. This is
+  something we could all get better at, so don't be shy in asking for what you need
+  to do great work :smile:.
+
+  See more on [asking for help](#asking-for-help) below!
+
+- [`help wanted` issues](https://github.com/weaveworks/eksctl/labels/help%20wanted)
+  are for those a little more familiar with the code base, but should still be accessible enough
+  to newcomers.
+
+- All other issues labelled `kind/<x>` or `priority/<x>` are also up for grabs, but
+  are likely to require a fair amount of context.
+
+## Developing eksctl
+
+**Sections:**
+- [Getting started](#getting-started)
+- [Setting up your Go environment](#setting-up-your-go-environment)
+- [Forking the repo](#forking-the-repo)
+- [Building eksctl](#building-eksctl)
+- [Running the unit tests](#running-the-unit-tests)
+- [Running the integration tests](#running-the-integration-tests)
+- [Writing your solution](#writing-your-solution)
```

Comment:
shall we note we've only tested these commands on MacOS & Linux? I'm guessing the majority of this doesn't work on Windows, although maybe with the new linux subsystem it might not be too far away from working.

Reply:
There is a note somewhere that the Makefile does not work on windows, but yet i will add that note too